libqtile/backend/x11/core.py

```diff
@@ -418,11 +418,24 @@ def update_client_list(self, windows_map: dict[int, base.WindowType]) -> None:
         This is needed for third party tasklists and drag and drop of tabs in
         chrome
         """
+        assert self.qtile
         # Regular top-level managed windows, i.e. excluding Static, Internal and Systray Icons
-        wids = [wid for wid, c in windows_map.items() if isinstance(c, window.Window)]
+        wids = [wid for wid, c in self.qtile.windows_map.items() if isinstance(c, window.Window)]
         self._root.set_property("_NET_CLIENT_LIST", wids)
-        # TODO: check stack order
-        self._root.set_property("_NET_CLIENT_LIST_STACKING", wids)
+
+    def update_client_stack(self) -> None:
+        """Set the current stacking order of clients to the given list of windows"""
+        assert self.qtile
+        stack = list(self._root.query_tree())
+        wids = [
+            wid
+            for wid, c in self.qtile.windows_map.items()
+            if isinstance(c, window.Window) and c.group and wid in stack
+        ]
+        # We need to keep the order in the tree so pick the items from that list
+        # but only keep the ones filtered out by wids
+        current_stack = [s for s in stack if s in wids]
+        self._root.set_property("_NET_CLIENT_LIST_STACKING", current_stack)
```

Comment:
I think passing `wids` to `set_property` would have the same effect, right? `wids` is filtered by `stack` in the list comprehension, and we ignore those items in stack that aren't also in `self.qtile.windows_map`, so `wids` is already the intersection between `stack` and `self.qtile.windows_map.keys()`. So `current_stack == wids`.

Reply:
Did you make this change and not yet push? just wondering if it was in fact resolved